pykafka/simpleconsumer.py

```diff
@@ -828,7 +829,18 @@ def enqueue_messages(self, messages):
         :type messages: Iterable of :class:`pykafka.common.Message`
         """
         for message in messages:
-            if message.offset != self.next_offset:
+            # HAck to deal with compacted topic not returning earliest offset
+            # correctly
+            if not self._consumed_first_msg and message.offset != self.next_offset:
+                log.info("Detected compacted topic. "
+                          "Setting earliest offest to (%s) "
+                          "not equal to next_offset (%s)",
+                          message.offset, self.next_offset)
+                self.next_offset = message.offset
+                self._consumed_first_msg = True
```

Comment:
I originally put the `_consumed_first_msg` in there because kafka was always reporting 0 as the earliest offset. (and i didn't notice the non-incrementing nature of the offsets at first).

I guess as long as the consumer will accept a message that has a larger offset this is not needed.